src/averageAges.js

```diff
@@ -37,7 +48,18 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let women = people.filter(person => person.sex === 'f');
+
+  if (withChildren) {
+    women = women.filter(woman =>
+      people.some(person => person.mother === woman.name));
+  }
+
+  const sumOfAges = women
+    .map(woman => woman.died - woman.born)
+    .reduce((sumOfAge, prevSum) => sumOfAge + prevSum);
+
+  return sumOfAges / women.length;
```

Comment:
You use `reduce` and division by length in every function. Maybe we should move this logic to a separate helper?